ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -247,13 +248,13 @@ public void addRaftPeers(Collection<RaftPeer> newPeers) {
     proxies.addPeers(newPeers);
   }
 
-  private StreamInfo newStreamInfo(ByteBuf buf) {
+  private StreamInfo newStreamInfo(ByteBuf buf, boolean isPrimary) {
     try {
       final RaftClientRequest request = ClientProtoUtils.toRaftClientRequest(
           RaftClientRequestProto.parseFrom(buf.nioBuffer()));
       final StateMachine stateMachine = server.getStateMachine(request.getRaftGroupId());
       return new StreamInfo(request, stateMachine.data().stream(request),
-          proxies.getDataStreamOutput(request));
+          isPrimary ? proxies.getDataStreamOutput(request) : Collections.EMPTY_LIST);
```

Comment:
It is a good idea.   We may use
```
      final List<DataStreamOutputRpc> outs = server.getId().equals(request.getServerId())?  
          proxies.getDataStreamOutput(request): Collections.emptyList();
      return new StreamInfo(request, stateMachine.data().stream(request), outs);
```
Note that request is RaftClientRequest.

Reply:
Ack. Didn't notice that the RaftClientRequest in header has the server id.

Now this PR becomes very simple again :)